packages/jest-worker/src/workers/messageParent.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {isMainThread, parentPort} from 'worker_threads';
+import {PARENT_MESSAGE_CUSTOM} from '../types';
+
+const isWorkerThread = () => {
```

Comment:
Can just be `const isWorkerThread = !isMainThread && parentPort`. If we want it to be lazy, at least ditch the try-catch